mq/3/reference/dlq/index.md

```diff
@@ -0,0 +1,146 @@
+---
+title: IronMQ Dead Letter Queue Reference
+layout: default
+section: mq-v3
+breadcrumbs:
+  - ['Reference', '/reference']
+  - ['Dead Letter Queue', '/dlq']
+---
+
+<section id="toc">
+  <h3>Table of Contents</h3>
+  <ul>
+    <li><a href="#overview">Overview</a></li>
+    <li><a href="#creating">Creating a Dead Letter Queue</a></li>
+    <li><a href="#deleting">Deleting a Dead Letter Queue</a></li>
+    <li><a href="#behavior">Behavior</a></li>
+    <li><a href="#use-case">One Time Delivery Use Case</a></li>
+  </ul>
+</section>
+
+<h2 id="overview">Overview</h2>
+
+Dead Letter Queues are queues that can be used to send messages to that for
+whatever reason could not be successfully processed on the source queue.
+Messages that have been attempted to be processed at least a specified number
+of times on the source queue will be placed onto the specified dead letter
+queue. The messages on the dead letter queue can then be processed again or
+the user can later try to determine why they weren't successfully processed in the
+first place, without having to throw them away.
+
+<h2 id="creating">Creating a Dead Letter Queue</h2>
+
+To add a dead letter queue to a queue, use the create or update queue
+endpoint:
+
+```
+PUT /3/projects/:project/queues/:queue
+
+{"queue":{"dead_letter":{"queue_name":"deadies", "max_reservations":1}}}
+```
+
+`"queue_name"` must be a non-empty string in order to create a dead letter
+queue, otherwise the request will error unless the queue already has a dead
+letter queue defined on it. `"max_reservations"` can be specified if desired
+to be a value `1 <= max_reservations <= 1000`, and will default to 10 if
+omitted. A dead letter queue cannot be specified on a push queue. The queue
+specified to be the dead letter queue can already exist or can be a queue that
+doesn't yet exist and we will create it when necessary.
+
+<h2 id="deleting">Removing a Dead Letter Queue</h2>
+
+In order to turn off the dead letter queue feature on a queue, the user should
+use the update queue endpoint with an empty `"queue_name"` in the
+`"dead_letter"` field:
+
+```
+PUT /3/projects/:projects/queues/:queue
+
+{"queue":{"dead_letter":{"queue_name":""}}}
+```
+
+The response will be a queue without a `"dead_letter"` section, if successful.
+This action will not remove the messages on the queue or delete the queue previously
+at `"queue_name"`. The only consequence of this action is that messages will
+no longer be sent from the source queue to the dead letter queue.
+
+<h2 id="behavior">Behavior</h2>
+
+Upon creation of a dead letter queue, any message, including any existing
+messages on the queue, whose `reserved_count` field exceeds the
+`dead_letter.max_reservations` threshold will be placed onto the queue
+specified as `dead_letter.queue_name` under the same project, and removed from
+the current queue. If this queue already exists, we will put the messages onto
```

Comment:
Instead of the "If this queue already exists" sentence, maybe:

If a queue with the given name already exists, it's used as is. Otherwise, a new queue is automatically created with the default pull queue settings.

(Add clarification in response to my question below as needed.)